server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -688,4 +693,14 @@ private static void validateNodeBandwidthRecoverySettings(Settings settings) {
             );
         }
     }
+
+    /**
+     * Whether the node bandwidth recovery settings are set.
+     */
+    public static boolean hasNodeBandwidthRecoverySettings(Settings settings) {
+        return NODE_BANDWIDTH_RECOVERY_SETTINGS.stream()
+            .filter(setting -> setting.get(settings) != ByteSizeValue.MINUS_ONE)
```

Comment:
I copied this logic from the existing `validateNodeBandwidthRecoverySettings()` function above. I see the node bandwidth settings currently have as default -1 and I see they allow to be configured to -1. Thus .exists() probably is not best, since one could still theoretically set all 3 node bandwidth settings explicitly to -1 to disable them. cc @tlrx to confirm my thinking or comment.

Reply:
Sure, we can keep this for now then.